### PR TITLE
Fixing missing theming variables

### LINKED ISSUE #3249: Highlighted counter is broken on dark mode
Items are:
* Outlined
* Highlighted
* Normal

Bright | Dark
---|---
Bildschirmfoto vom 2022-09-14 09-48-33 | Bildschirmfoto vom 2022-09-14 09-48-23

The difference between backgrounds of highlighted and normal is non-existant and especially the missing contrast of the font on normal is an issue

### PR DESCRIPTION
**Pull request by skjnldsv**
fix https://github.com/nextcloud/nextcloud-vue/issues/3249

Element primary is based on mainBackground, so it needs to be re-calculated for the dark theme too

| Before | After |
|:---------:|:------:|
|image|image|

Here is a diff between the two highcontrast themes for easier review
image

### CODE REVIEW

File: apps/theming/lib/Themes/DefaultTheme.php [hunk 1]
```diff
@@ -238,6 +238,7 @@ public function getCSSVariables(): array {
 			$variables["--image-logoheader-custom"] = 'true';
 		}
 
+
```
Comment (nickvergessen) [inline apps/theming/lib/Themes/DefaultTheme.php:241]:
```suggestion
```

**Review by nickvergessen — APPROVED** [review]

**Comment by skjnldsv** [comment]
> @skjnldsv do we need to do this for HighContrastDark as well?

Should not as the `HighContrastDarkTheme` is extending from `DarkTheme`

File: apps/theming/lib/Themes/DefaultTheme.php [hunk 2]
```diff
@@ -248,6 +249,8 @@ public function getCSSVariables(): array {
 				$variables['--image-main-background'] = "url('" . $this->urlGenerator->linkToRouteAbsolute('theming.theming.getBackground') . "')";
 			} elseif ($themingBackground !== 'default' && substr($themingBackground, 0, 1) !== '#') {
 				$variables['--image-main-background'] = "url('" . $this->urlGenerator->linkTo(Application::APP_ID, "/img/background/$themingBackground") . "')";
+			} elseif (substr($themingBackground, 0, 1) === '#') {
```
Comment (skjnldsv) [inline apps/theming/lib/Themes/DefaultTheme.php:252]: AH damn, that's a leftover

**Review by ChristophWurst — APPROVED** [review]
:+1: :elephant:

File: apps/theming/lib/Themes/DefaultTheme.php (hunk 2 above)
Comment (skjnldsv) [reply apps/theming/lib/Themes/DefaultTheme.php]: Fixed

**Comment by skjnldsv** [comment]
> @skjnldsv do we need to do this for HighContrastDark as well?

Hum, actually yes, we should because the background and text changes slightly

File: apps/theming/lib/Themes/CommonThemeTrait.php [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+trait CommonThemeTrait {
+	public Util $util;
+
+	/**
+	 * Generate primary-related variables
+	 * This is shared between multiple themes because colorMainBackground and colorMainText
+	 * will change in between.
+	 */
+	protected function genPrimaryVariables(string $colorMainBackground, string $colorMainText): array {
```
Comment (nickvergessen) [inline apps/theming/lib/Themes/CommonThemeTrait.php:37]:
```suggestion
	protected function generatePrimaryVariables(string $colorMainBackground, string $colorMainText): array {
```

we wouldn't need to save characters like this

Comment (skjnldsv) [reply apps/theming/lib/Themes/CommonThemeTrait.php]: Done

**Pull request merged**